golem-common/src/model/mod.rs

```diff
@@ -2482,6 +2483,243 @@ impl FromStr for ComponentType {
     }
 }
 
+/// Key that can be used to identify a component file.
+/// All files with the same content will have the same key.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, NewType)]
+pub struct InitialComponentFileKey(pub String);
```

Comment:
The design that I went for is that all initial are stored based on their hash in the blob storage.
This way we can have good sharing of data without making the implementation more complex.

The hashes are computed server-side so we are resistant to any attacks.
Collisions for sha-256 should basically not exist in practice.